tests/trainer/test_trainer.py

```diff
@@ -3769,3 +3769,37 @@ def test_hyperparameter_search_backends(self):
             list(ALL_HYPERPARAMETER_SEARCH_BACKENDS.keys()),
             list(HPSearchBackend),
         )
+
+
+class OptimizerAndModelInspectionTest(unittest.TestCase):
+    def test_get_num_trainable_parameters(self):
+        in_features = 128
+        out_features = 64
+        # in_features * out_features + bias
+        expected_num_params = in_features * out_features + out_features
+        model = nn.Sequential(nn.Linear(in_features, out_features))
```

Comment:
Let's perhaps make this two linear layers, one that's frozen, one that's not, so this way trainable checks if we have different optimizer groups? :)

Reply:
Yes, you're right, that would definitely be better :)